packages/fuel-gauge/src/payable-annotation.test.ts

```diff
@@ -39,7 +39,7 @@ test("don't allow sending coins to non-payable functions", async () => {
   const baseAssetId = contract.provider.getBaseAssetId();
 
   // This should fail because the function is not payable
-  await expect(async () =>
+  await expect(() =>
```

Comment:
Perhaps to make this more intuitive we can refactor as such

```suggestion
expect(async () => {
    const tx = await contract.functions
      .non_payable()
      .callParams({
        forward: {
          amount: bn(100),
          assetId: baseAssetId,
        },
      })
      .call();

    await tx.waitForResult();
  }).rejects.toThrowError(
    `The target function non_payable cannot accept forwarded funds as it's not marked as 'payable'.`
  );
   ```

Reply:
The `await` should not have been removed.

Reply:
Why is that? If we are awaiting the contract call isn't it the same result?